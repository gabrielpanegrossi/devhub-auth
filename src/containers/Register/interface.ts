export interface Values {
  name: string;
  lastName: string;
  email: string;
  password: string;
  passwordConfirmation: string;
}
