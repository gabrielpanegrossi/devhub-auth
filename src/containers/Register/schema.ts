import * as yup from 'yup';
import { auth } from '~services';
import { validateEmail } from '~utils';

export const validationSchema = yup.object({
  name: yup.string().required('Please insert your Name.'),
  lastName: yup.string().required('Please insert your Last Name.'),
  email: yup
    .string()
    .email('Please insert a valid email.')
    .required('Please insert an email.')
    .test('checkEmailUnique', 'This email is already registered.', async function (value) {
      if (validateEmail(value)) {
        const response = await auth.emailExists(value || '');
        if (response) return response.exists;
      }
      return false;
    }),
  password: yup
    .string()
    .min(8, 'The password must have at least 8 characters.')
    .required('Please insert a password.'),
  passwordConfirmation: yup
    .string()
    .required('Please confirm your password.')
    .test('match', 'Passwords do not match.', function () {
      return this.parent.password === this.parent.passwordConfirmation;
    }),
});
