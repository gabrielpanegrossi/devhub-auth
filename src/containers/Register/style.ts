import styled from 'styled-components';
import { Text as DefaultText, Link as DefaultLink, Subtitle as DefaultSubtitle } from '~components';

export const Container = styled.section`
  display: flex;
  padding: 16px;
  height: 100%;
  background: #f2f4f7;

  @media (min-width: 1024px) {
    align-items: center;
  }
`;

export const Content = styled.div`
  display: flex;
  align-items: center;
  height: 100%;
  width: 100%;
  max-width: 1080px;
  margin: 0 auto;

  @media (min-width: 1024px) {
    display: flex;
    justify-content: center;
  }
`;

export const Handler = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  max-width: 448px;
  margin: 0 auto;
  overflow: hidden;

  > section {
    background: #3c444c;
    border-radius: 5px;
  }

  @media (min-width: 1024px) {
    width: 50%;
  }
`;

export const Register = styled.section`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
`;

export const Auth = styled.section`
  display: flex;
  justify-content: center;
  width: 100%;
  padding: 24px;
  margin-top: 24px;

  > p {
    color: #fff;
    margin-right: 8px;
  }
`;

export const Subtitle = styled(DefaultSubtitle)`
  width: fit-content;
  margin: 16px auto 0;
  color: #fff;
  letter-spacing: 5px;
  text-transform: uppercase;
`;

export const Text = styled(DefaultText)`
  letter-spacing: 5px;
  width: fit-content;
  margin: 0 auto;

  @media (min-width: 1024px) {
    margin: 0;
  }
`;

export const SubText = styled(DefaultText)`
  color: #fff;
`;

export const Link = styled(DefaultLink)`
  margin-top: 16px;
`;
