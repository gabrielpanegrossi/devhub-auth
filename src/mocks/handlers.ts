import registerHandler from './handlers/register';
import signinHandler from './handlers/auth';
import validateEmailHandler from './handlers/emailExists';
import validateTokenHandler from './handlers/validateAuthorization';
import validatePasswordRecoveryCode from './handlers/validatePasswordRecoveryCode';

export const handlers = [
  ...registerHandler,
  ...signinHandler,
  ...validateEmailHandler,
  ...validateTokenHandler,
  ...validatePasswordRecoveryCode,
];
