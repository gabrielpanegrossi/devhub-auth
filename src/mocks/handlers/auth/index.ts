import { rest } from 'msw';
import { Payload } from './interface';

const Handler = [
  rest.post<Payload>(`${process.env.REACT_APP_BACKEND_URL}/signin`, (req, res, ctx) => {
    const payload = req.body;
    if (payload.email === 'gabriel@gmail.com')
      return res(
        ctx.delay(1000),
        ctx.status(200),
        ctx.json({
          token: '6B810C90AA9A99858230C3D5ED479096A190FA4B',
        })
      );

    return res(
      ctx.delay(1000),
      ctx.status(401),
      ctx.json({ errorMessage: 'email or password incorrect!' })
    );
  }),
];

export default Handler;
