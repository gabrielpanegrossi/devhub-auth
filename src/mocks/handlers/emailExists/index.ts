import { rest } from 'msw';
import { Payload } from './interface';

const Handler = [
  rest.post<Payload>(`${process.env.REACT_APP_BACKEND_URL}/email-exists`, (req, res, ctx) => {
    const payload = req.body;
    if (payload.email === 'gabriel@gmail.com')
      return res(ctx.delay(1000), ctx.status(200), ctx.json({ status: 'success', exists: true }));

    return res(ctx.delay(1000), ctx.status(200), ctx.json({ status: 'error', exists: false }));
  }),
];

export default Handler;
