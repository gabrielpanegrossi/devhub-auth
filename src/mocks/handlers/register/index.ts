import { rest } from 'msw';
import { Payload } from './interface';

const Handler = [
  rest.post<Payload>(`${process.env.REACT_APP_BACKEND_URL}/register`, (req, res, ctx) => {
    if (req.body.email === 'gabriel@gmail.com')
      return res(
        ctx.delay(1000),
        ctx.status(200),
        ctx.json({ status: 'success', message: `User created!` })
      );

    return res(
      ctx.delay(1000),
      ctx.status(200),
      ctx.json({ status: 'error', message: `Try again later.` })
    );
  }),
];

export default Handler;
