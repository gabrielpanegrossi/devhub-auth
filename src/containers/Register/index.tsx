import React from 'react';
import { useRequest } from 'ahooks';
import { useNavigate } from 'react-router-dom';
import { Form, Field, Button, Text, Link } from '~components';
import { auth } from '~services';
import { Values } from './interface';
import { validationSchema } from './schema';
import * as Styled from './style';

function Register() {
  const initialValues = {
    name: '',
    lastName: '',
    email: '',
    password: '',
    passwordConfirmation: '',
  };
  const navigate = useNavigate();

  const { runAsync, loading } = useRequest(auth.register, { manual: true });

  const handleSubmit = async (values: Values) => {
    console.log('primeiro');
    await runAsync(values);
    console.log('segundo');
    navigate('/');
  };

  return (
    <Styled.Container>
      <Styled.Content>
        <Styled.Handler>
          <Styled.Register>
            <Styled.Subtitle id='register-form-title'>Sign Up</Styled.Subtitle>
            <Styled.SubText>Access all your dev content on the same place!</Styled.SubText>
            <Form
              onSubmit={handleSubmit}
              initialValues={initialValues}
              validationSchema={validationSchema}
              aria-labelledby='register-form-title'
            >
              {({ values }: { values: Values }) => {
                return (
                  <>
                    <Field name='name' label='Name *' value={values.name} autoFocus={true} />
                    <Field name='lastName' label='Last Name *' value={values.lastName} />
                    <Field name='email' label='Email *' value={values.email} />
                    <Field
                      name='password'
                      label='Password *'
                      type='password'
                      value={values.password}
                    />
                    <Field
                      name='passwordConfirmation'
                      label='Confirm your password *'
                      type='password'
                      value={values.passwordConfirmation}
                    />
                    <Button type='submit' loading={loading}>
                      Sign Up
                    </Button>
                  </>
                );
              }}
            </Form>
          </Styled.Register>
          <Styled.Auth>
            <Text>Do you have an account?</Text>
            <Link to='/'>Log in</Link>
          </Styled.Auth>
        </Styled.Handler>
      </Styled.Content>
    </Styled.Container>
  );
}

export default Register;
