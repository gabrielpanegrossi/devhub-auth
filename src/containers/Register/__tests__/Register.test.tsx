import { render, screen, waitFor } from '@testing-library/react';
import { setupServer } from 'msw/node';
import * as MockReactRouterDom from 'react-router-dom';
import { handlers } from '~mocks/handlers';
import { BrowserRouter as Router } from 'react-router-dom';
import Register from '../index';
import userEvent from '@testing-library/user-event';

jest.setTimeout(15000);

const server = setupServer(...handlers);
const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual<typeof MockReactRouterDom>('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('Register flow', () => {
  beforeEach(() => {
    render(
      <Router>
        <Register />
      </Router>
    );
  });
  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());

  it('should match snpashot', () => {
    const container = render(
      <Router>
        <Register />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render', () => {
    expect(screen.queryByRole('form')).toBeInTheDocument();
    expect(screen.queryByRole('button')).toBeEnabled();
  });

  it('should validate all form fields', async () => {
    const button = screen.queryByRole('button');

    if (button) userEvent.click(button);

    await waitFor(() => {
      expect(screen.queryAllByLabelText('Input error').length).toBe(5);
    });
  });

  it('should submit form and navigate to next page', async () => {
    const user = userEvent.setup();

    const button = screen.queryByRole('button');
    const nameField = screen.queryByLabelText('name');
    const lastNameField = screen.queryByLabelText('lastName');
    const emailField = screen.queryByLabelText('email');
    const passwordField = screen.queryByLabelText('password');
    const confirmPasswordField = screen.queryByLabelText('passwordConfirmation');

    await user.type(nameField, 'Gabriel');
    await user.type(lastNameField, 'Santos');
    await user.type(emailField, 'gabriel@gmail.com');
    await user.type(passwordField, 'salada123');
    await user.type(confirmPasswordField, 'salada123');

    await user.click(button);

    await waitFor(() => {
      expect(mockNavigate).toBeCalledWith('/');
    });
  });
});
