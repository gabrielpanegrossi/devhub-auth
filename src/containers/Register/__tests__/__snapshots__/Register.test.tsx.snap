// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register flow should match snpashot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c9 {
  width: 100%;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  font-size: 10px;
  color: white;
  text-transform: uppercase;
}

.c11 {
  width: 100%;
  height: 40px;
  margin-top: 8px;
  border-radius: 3px;
  outline: transparent;
  padding: 0 8px;
}

.c13 {
  position: relative;
  width: 100%;
  height: 40px;
  border-radius: 10px;
  font-size: 10px;
  cursor: pointer;
  color: #fff;
  border: 1px solid #fff;
  background: #A154F2;
  text-transform: uppercase;
  font-size: 14px;
  -webkit-letter-spacing: 2.5px;
  -moz-letter-spacing: 2.5px;
  -ms-letter-spacing: 2.5px;
  letter-spacing: 2.5px;
  font-weight: bold;
  -webkit-transition: background 0.5s;
  transition: background 0.5s;
}

.c13:hover {
  background: #F27A54;
}

.c13:before {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  content: ""attr(data-content)"";
  width: 100%;
  height: 100%;
  top: -1px;
  left: -1px;
  background: #F27A54;
  border: 1px solid #fff;
  border-radius: 10px;
  -webkit-transition: opacity 2s;
  transition: opacity 2s;
  opacity: 0;
  z-index: 1;
}

.c13:hover::before {
  opacity: 1;
  background: linear-gradient(90deg,#F27A54 0%,#A154F2 100%);
}

.c13:disabled {
  opacity: 0.5;
}

.c12 {
  position: relative;
  width: 100%;
  max-width: 285px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  padding: 24px;
}

.c8 > :not(:first-child) {
  margin-top: 24px;
}

.c4 {
  font-size: 32px;
  margin: 0;
  padding: 0;
}

.c6 {
  margin: 0;
  padding: 0;
  font-size: 14px;
}

.c15 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c15:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 16px;
  height: 100%;
  background: #f2f4f7;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  max-width: 1080px;
  margin: 0 auto;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  max-width: 448px;
  margin: 0 auto;
  overflow: hidden;
}

.c2 > section {
  background: #3c444c;
  border-radius: 5px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  padding: 24px;
  margin-top: 24px;
}

.c14 > p {
  color: #fff;
  margin-right: 8px;
}

.c5 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  margin: 16px auto 0;
  color: #fff;
  -webkit-letter-spacing: 5px;
  -moz-letter-spacing: 5px;
  -ms-letter-spacing: 5px;
  letter-spacing: 5px;
  text-transform: uppercase;
}

.c7 {
  color: #fff;
}

@media (min-width:1024px) {
  .c0 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (min-width:1024px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (min-width:1024px) {
  .c2 {
    width: 50%;
  }
}

<body>
    <div>
      <section
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <section
              class="c3"
            >
              <h2
                class="c4 c5"
              >
                Sign Up
              </h2>
              <p
                class="c6 c7"
              >
                Access all your dev content on the same place!
              </p>
              <form
                action="#"
                aria-labelledby="register-form-title"
                class="c8"
              >
                <div
                  class="c9"
                >
                  <label
                    class="c10"
                  >
                    Name *
                    <input
                      aria-label="name"
                      class="c11"
                      name="name"
                      value=""
                    />
                  </label>
                </div>
                <div
                  class="c9"
                >
                  <label
                    class="c10"
                  >
                    Last Name *
                    <input
                      aria-label="lastName"
                      class="c11"
                      name="lastName"
                      value=""
                    />
                  </label>
                </div>
                <div
                  class="c9"
                >
                  <label
                    class="c10"
                  >
                    Email *
                    <input
                      aria-label="email"
                      class="c11"
                      name="email"
                      value=""
                    />
                  </label>
                </div>
                <div
                  class="c9"
                >
                  <label
                    class="c10"
                  >
                    Password *
                    <input
                      aria-label="password"
                      class="c11"
                      name="password"
                      type="password"
                      value=""
                    />
                  </label>
                </div>
                <div
                  class="c9"
                >
                  <label
                    class="c10"
                  >
                    Confirm your password *
                    <input
                      aria-label="passwordConfirmation"
                      class="c11"
                      name="passwordConfirmation"
                      type="password"
                      value=""
                    />
                  </label>
                </div>
                <div
                  class="c12"
                >
                  <button
                    class="c13"
                    data-content="Sign Up"
                    type="submit"
                  >
                    Sign Up
                  </button>
                </div>
              </form>
            </section>
            <section
              class="c14"
            >
              <p
                class="c6"
              >
                Do you have an account?
              </p>
              <a
                class="c15"
                href="/"
              >
                Log in
              </a>
            </section>
          </div>
        </div>
      </section>
    </div>
    <div>
      <section
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <section
              class="c3"
            >
              <h2
                class="c4 c5"
              >
                Sign Up
              </h2>
              <p
                class="c6 c7"
              >
                Access all your dev content on the same place!
              </p>
              <form
                action="#"
                aria-labelledby="register-form-title"
                class="c8"
              >
                <div
                  class="c9"
                >
                  <label
                    class="c10"
                  >
                    Name *
                    <input
                      aria-label="name"
                      class="c11"
                      name="name"
                      value=""
                    />
                  </label>
                </div>
                <div
                  class="c9"
                >
                  <label
                    class="c10"
                  >
                    Last Name *
                    <input
                      aria-label="lastName"
                      class="c11"
                      name="lastName"
                      value=""
                    />
                  </label>
                </div>
                <div
                  class="c9"
                >
                  <label
                    class="c10"
                  >
                    Email *
                    <input
                      aria-label="email"
                      class="c11"
                      name="email"
                      value=""
                    />
                  </label>
                </div>
                <div
                  class="c9"
                >
                  <label
                    class="c10"
                  >
                    Password *
                    <input
                      aria-label="password"
                      class="c11"
                      name="password"
                      type="password"
                      value=""
                    />
                  </label>
                </div>
                <div
                  class="c9"
                >
                  <label
                    class="c10"
                  >
                    Confirm your password *
                    <input
                      aria-label="passwordConfirmation"
                      class="c11"
                      name="passwordConfirmation"
                      type="password"
                      value=""
                    />
                  </label>
                </div>
                <div
                  class="c12"
                >
                  <button
                    class="c13"
                    data-content="Sign Up"
                    type="submit"
                  >
                    Sign Up
                  </button>
                </div>
              </form>
            </section>
            <section
              class="c14"
            >
              <p
                class="c6"
              >
                Do you have an account?
              </p>
              <a
                class="c15"
                href="/"
              >
                Log in
              </a>
            </section>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": .c9 {
  width: 100%;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  font-size: 10px;
  color: white;
  text-transform: uppercase;
}

.c11 {
  width: 100%;
  height: 40px;
  margin-top: 8px;
  border-radius: 3px;
  outline: transparent;
  padding: 0 8px;
}

.c13 {
  position: relative;
  width: 100%;
  height: 40px;
  border-radius: 10px;
  font-size: 10px;
  cursor: pointer;
  color: #fff;
  border: 1px solid #fff;
  background: #A154F2;
  text-transform: uppercase;
  font-size: 14px;
  -webkit-letter-spacing: 2.5px;
  -moz-letter-spacing: 2.5px;
  -ms-letter-spacing: 2.5px;
  letter-spacing: 2.5px;
  font-weight: bold;
  -webkit-transition: background 0.5s;
  transition: background 0.5s;
}

.c13:hover {
  background: #F27A54;
}

.c13:before {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  content: ""attr(data-content)"";
  width: 100%;
  height: 100%;
  top: -1px;
  left: -1px;
  background: #F27A54;
  border: 1px solid #fff;
  border-radius: 10px;
  -webkit-transition: opacity 2s;
  transition: opacity 2s;
  opacity: 0;
  z-index: 1;
}

.c13:hover::before {
  opacity: 1;
  background: linear-gradient(90deg,#F27A54 0%,#A154F2 100%);
}

.c13:disabled {
  opacity: 0.5;
}

.c12 {
  position: relative;
  width: 100%;
  max-width: 285px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  padding: 24px;
}

.c8 > :not(:first-child) {
  margin-top: 24px;
}

.c4 {
  font-size: 32px;
  margin: 0;
  padding: 0;
}

.c6 {
  margin: 0;
  padding: 0;
  font-size: 14px;
}

.c15 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c15:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 16px;
  height: 100%;
  background: #f2f4f7;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  max-width: 1080px;
  margin: 0 auto;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  max-width: 448px;
  margin: 0 auto;
  overflow: hidden;
}

.c2 > section {
  background: #3c444c;
  border-radius: 5px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  padding: 24px;
  margin-top: 24px;
}

.c14 > p {
  color: #fff;
  margin-right: 8px;
}

.c5 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  margin: 16px auto 0;
  color: #fff;
  -webkit-letter-spacing: 5px;
  -moz-letter-spacing: 5px;
  -ms-letter-spacing: 5px;
  letter-spacing: 5px;
  text-transform: uppercase;
}

.c7 {
  color: #fff;
}

@media (min-width:1024px) {
  .c0 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (min-width:1024px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (min-width:1024px) {
  .c2 {
    width: 50%;
  }
}

<div>
    <section
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <section
            class="c3"
          >
            <h2
              class="c4 c5"
            >
              Sign Up
            </h2>
            <p
              class="c6 c7"
            >
              Access all your dev content on the same place!
            </p>
            <form
              action="#"
              aria-labelledby="register-form-title"
              class="c8"
            >
              <div
                class="c9"
              >
                <label
                  class="c10"
                >
                  Name *
                  <input
                    aria-label="name"
                    class="c11"
                    name="name"
                    value=""
                  />
                </label>
              </div>
              <div
                class="c9"
              >
                <label
                  class="c10"
                >
                  Last Name *
                  <input
                    aria-label="lastName"
                    class="c11"
                    name="lastName"
                    value=""
                  />
                </label>
              </div>
              <div
                class="c9"
              >
                <label
                  class="c10"
                >
                  Email *
                  <input
                    aria-label="email"
                    class="c11"
                    name="email"
                    value=""
                  />
                </label>
              </div>
              <div
                class="c9"
              >
                <label
                  class="c10"
                >
                  Password *
                  <input
                    aria-label="password"
                    class="c11"
                    name="password"
                    type="password"
                    value=""
                  />
                </label>
              </div>
              <div
                class="c9"
              >
                <label
                  class="c10"
                >
                  Confirm your password *
                  <input
                    aria-label="passwordConfirmation"
                    class="c11"
                    name="passwordConfirmation"
                    type="password"
                    value=""
                  />
                </label>
              </div>
              <div
                class="c12"
              >
                <button
                  class="c13"
                  data-content="Sign Up"
                  type="submit"
                >
                  Sign Up
                </button>
              </div>
            </form>
          </section>
          <section
            class="c14"
          >
            <p
              class="c6"
            >
              Do you have an account?
            </p>
            <a
              class="c15"
              href="/"
            >
              Log in
            </a>
          </section>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
